Build fails on illumos with duplicate definition of strnstr

On illumos (an OpenSolaris derivative), build fails with:

gcc -DHAVE_CONFIG_H -I.  -I. -I./include -I./src -I./src  -D_GNU_SOURCE -W -Wall -g -O -I/usr/include/pcre  -fPIE -DPIE -Wundef -Wmissing-format-attribute -Wshadow -Wlogical-op -MT src/attributes.o -MD -MP -MF $depbase.Tpo -c -o src/attributes.o src/attributes.c &&\
mv -f $depbase.Tpo $depbase.Po
In file included from src/attributes.c:25:0:
./include/util.h:64:7: error: conflicting types for 'strnstr'
 char* strnstr(char* haystack,char* needle,int n);
       ^~~~~~~
In file included from src/attributes.c:22:0:
/usr/include/string.h:125:14: note: previous declaration of 'strnstr' was here
 extern char *strnstr(const char *, const char *, size_t);
              ^~~~~~~

It's correctly detected that we do in fact have strnstr and put the correct value into config.h, but nothing in attributes.c or util.h includes config.h, so the definition of HAVE_STRNSTR is never picked up.

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,6 @@
+2021-11-14 Jason Pyeron <jpyeron@pdinc.us>
+	* Add missing config.h include (closes: #104)
+
 2021-11-07 Hannes von Haugwitz <hannes@vonhaugwitz.com>
 	* Remove bashism in configure.ac
 

--- a/include/util.h
+++ b/include/util.h
@@ -21,6 +21,7 @@
 
 #ifndef _UTIL_H_INCLUDED
 #define _UTIL_H_INCLUDED
+#include "config.h"
 #include <string.h>
 #include <stdbool.h>
 #include <sys/types.h>

